apiserver/Makefile

```diff
@@ -32,6 +35,10 @@ jupyter_notebook_config.py:
 	    -e "s,@keyfile@,$(GSA_KEY_FILE),g" \
             < jupyter_notebook_config.py.in > $@
 
+hailjwt: ../hailjwt/setup.py $(shell find ../hailjwt/hailjwt -iname \*.py)
+	cd ../ && rsync --relative $(?:../hailjwt/%=hailjwt/./%) apiserver/hailjwt
+	touch hailjwt
```

Comment:
what is the purpose of touching hailjwt here? if the copy fails, would this give false assurance?

Reply:
if the copy fails the touch is not executed. If I don't touch the directory, make always things the directory is out of date because the inode's modification time is older than the files.

Unfortunately, there's a race condition here that's hard to fix: if a file is added after `find` finds it but before I `touch`, then I won't realize the directory is out of sync. The best fix I can think of is to have find also print the modification time of the dependency inodes. Then I can set hailjwt to the max of those times. That significantly complicates this rule though (I'd have to parse the output of find so as not to include the timestamps themselves in the make dependencies) for a super rare case: someone is editing hailjwt at the same time as running make in the apiserver.

Reply:
I think making this PHONY fixes all these problems.